FocusOn/View/ProgressView.swift

```diff
@@ -6,15 +6,84 @@
 //
 
 import SwiftUI
+import SwiftUICharts
 
 struct ProgressView: View {
+    @StateObject var viewModel = ProgressViewModel()
+    
+    @State private var chartData: [DataPoint]?
+    
     var body: some View {
-        Text("Progress View")
+        VStack {
+            Text("Here is your progres")
+                .foregroundColor(.accentColor)
+                .font(.largeTitle)
+            ScrollView(.horizontal) {
+                LazyHStack(spacing: 20) {
+                    let chunkedChartData = chartData?.chunked(into: 7)
+                    /*
+                     ForEach(chartData) { data in
+                        BarChartView(dataPoints: data ?? [])
+                        .chartStyle(
+                            BarChartStyle(
+                                barMinHeight: 10,
+                                showAxis: false,
+                                showLegends: false
+                            )
+                        )
+                     }
+                     */
+                    ForEach(0..<(chunkedChartData?.count)!, id: \.self) {
+                        Text("Item \($0)")
+                            .foregroundColor(.accentColor)
+                            .font(.largeTitle)
+                        HStack {
+                            BarChartView(dataPoints: chartData ?? [])
+                                .chartStyle(
+                                    BarChartStyle(
+                                        barMinHeight: 10,
+                                        showAxis: false,
+                                        showLegends: false
+                                    )
+                                )
+                        }
+                        .padding(20)
+                    }
+                }
+            }
+            HStack(spacing: 20) {
+                BarChartView(dataPoints: chartData ?? [])
+                    .chartStyle(
+                        BarChartStyle(
+                            barMinHeight: 10,
+                            showAxis: false,
+                            showLegends: true
+                        )
+                    )
+            }
+            .padding(20)
+        }
+        .onAppear { fetchChartData() }
```

Comment:
As talked, the issue is here, this method is called after `let chunkedChartData = chartData?.chunked(into: 7)` ligne 23.